flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/TaskExecutorRegistration.java

```diff
@@ -72,13 +77,15 @@ public TaskExecutorRegistration(
 			final String taskExecutorAddress,
 			final ResourceID resourceId,
 			final int dataPort,
+			final int jmxPort,
 			final HardwareDescription hardwareDescription,
 			final TaskExecutorMemoryConfiguration memoryConfiguration,
 			final ResourceProfile defaultSlotResourceProfile,
 			final ResourceProfile totalResourceProfile) {
 		this.taskExecutorAddress = checkNotNull(taskExecutorAddress);
 		this.resourceId = checkNotNull(resourceId);
 		this.dataPort = dataPort;
+		this.jmxPort = jmxPort;
```

Comment:
I would prefer a more general approach; basically just a Map<String, String> containing various meta-data bits. Then we could easily add information about other services and/or ports in the future.

Reply:
we can do that, in fact it is very similar to the `JobConfigInfo` message content. 
do you want to move `dataPort` and some other components into this generic metadata collection? (we can mark them deprecated first then remove them in the next release)

Reply:
hmm...I think it would make sense to also include the `dataPort`; it doesn't appear to be used by the runtime? But that also seems to apply to the hardwareDescription and memoryConfiguration.

hmm...maybe we should instead just bundle them in container, that is then just passed along. It would achieve my goal of not having explicit accesses to the jmx Port in various components, while still allowing us to _use_ them for other things down the line.

Reply:
hmm. I understand your concern. and putting an opaque Map<String, String> would definitely be safer. 

could you elaborate what you meant by "just bundle them in container"? do you mean putting everything in the Map<String, String> container?

Reply:
> could you elaborate what you meant by "just bundle them in container"?

Just a POJO containing all these.

Reply:
after some experiment, it seems like it is modifying a whole lot of public classes including header/messages in REST API as well as TaskExecutor. I think separating that into a refactor PR might've been a better idea. what do you think?

Reply:
Let's leave it like it is then. I'm sure there is some way to achieve it, but it seems more hassle than it's worth.